File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.Utils;
using DevExpress.XtraGrid.Views.Grid;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            tbl.Columns.Add("Check", typeof(bool));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) });
            return tbl;
        }

        public Form1()
        {
            InitializeComponent();
            gridControl1.DataSource = CreateTable(20);
        }

        private void gridView1_MouseDown(object sender, MouseEventArgs e)
        {
            CheckGridViewClick(e, gridView1);
        }
        private static void CheckGridViewClick(MouseEventArgs e, GridView view)
        {
            GridHitInfo hi = view.CalcHitInfo(e.Location);
            if (hi.InRowCell)
            {
                GridViewInfo vi = view.GetViewInfo() as GridViewInfo;
                GridCellInfo cellInfo = vi.GetGridCellInfo(hi);
                if (cellInfo == null)
                    return;
                CheckEditViewInfo cInfo = cellInfo.ViewInfo as CheckEditViewInfo;
                if (cInfo == null)
                    return;
                Rectangle r = cInfo.CheckInfo.GlyphRect;
                r.Offset(cellInfo.Bounds.Location);
                if (r.Contains(e.Location))
                {
                    ProcessCheckClick(e, view, hi);
                }
            }
        }
        private static void ProcessCheckClick(MouseEventArgs e, GridView view, GridHitInfo hi)
        {
            view.FocusedColumn = hi.Column;
            view.FocusedRowHandle = hi.RowHandle;
            view.ShowEditor();
            CheckEdit edit = view.ActiveEditor as CheckEdit;
            if (edit == null) return;
            edit.Toggle();
            DXMouseEventArgs.GetMouseArgs(e).Handled = true;
        }

    }
}

File: VB/Form1.vb
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.Utils
Imports DevExpress.XtraGrid.Views.Grid

Namespace WindowsApplication1

    Public Partial Class Form1
        Inherits Form

        Private Function CreateTable(ByVal RowCount As Integer) As DataTable
            Dim tbl As DataTable = New DataTable()
            tbl.Columns.Add("Name", GetType(String))
            tbl.Columns.Add("ID", GetType(Integer))
            tbl.Columns.Add("Number", GetType(Integer))
            tbl.Columns.Add("Date", GetType(Date))
            tbl.Columns.Add("Check", GetType(Boolean))
            For i As Integer = 0 To RowCount - 1
                tbl.Rows.Add(New Object() {String.Format("Name{0}", i), i, 3 - i, Date.Now.AddDays(i)})
            Next

            Return tbl
        End Function

        Public Sub New()
            InitializeComponent()
            gridControl1.DataSource = CreateTable(20)
        End Sub

        Private Sub gridView1_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs)
            CheckGridViewClick(e, gridView1)
        End Sub

        Private Shared Sub CheckGridViewClick(ByVal e As MouseEventArgs, ByVal view As GridView)
            Dim hi As GridHitInfo = view.CalcHitInfo(e.Location)
            If hi.InRowCell Then
                Dim vi As GridViewInfo = TryCast(view.GetViewInfo(), GridViewInfo)
                Dim cellInfo As GridCellInfo = vi.GetGridCellInfo(hi)
                If cellInfo Is Nothing Then Return
                Dim cInfo As CheckEditViewInfo = TryCast(cellInfo.ViewInfo, CheckEditViewInfo)
                If cInfo Is Nothing Then Return
                Dim r As Rectangle = cInfo.CheckInfo.GlyphRect
                r.Offset(cellInfo.Bounds.Location)
                If r.Contains(e.Location) Then
                    ProcessCheckClick(e, view, hi)
                End If
            End If
        End Sub

        Private Shared Sub ProcessCheckClick(ByVal e As MouseEventArgs, ByVal view As GridView, ByVal hi As GridHitInfo)
            view.FocusedColumn = hi.Column
            view.FocusedRowHandle = hi.RowHandle
            view.ShowEditor()
            Dim edit As CheckEdit = TryCast(view.ActiveEditor, CheckEdit)
            If edit Is Nothing Then Return
            edit.Toggle()
            DXMouseEventArgs.GetMouseArgs(e).Handled = True
        End Sub
    End Class
End Namespace
